### ISSUE TITLE
Kernel compilation error

### ISSUE THREAD

**Issue by thmsmlt** [ctag-fh-kiel/ctag-face-2-4#1]
I'm trying to compile the kernel for a Beaglebone Black Wireless on a Ubuntu 14.04 machine. When trying to compile the kernel I get this error:

Makefile:660: Cannot use CONFIG_CC_STACKPROTECTOR_STRONG: -fstack-protector-strong not supported by compiler

Any ideas on what is going wrong?

I've also tried the quick setup SD image, but my Beaglebone Black Wireless won't boot with that image...

**Comment by henrix** [comment]
Did you use a recent version of Linaro compiler (see [here](https://github.com/ctag-fh-kiel/ctag-face-2-4/blob/master/docs/Kernel_Compilation_Guide.md))?
Furthermore, the driver is already included in the official BeagleBoard.org images, so you don't have to compile the kernel on yourself.

**Comment by thmsmlt** [comment]
I followed the steps in the guide and tried the Linaro install step again, the second command does not give any output from terminal, but that doesn't seem out of order to me. I'll attach the full error output to this message.

[output.txt](https://github.com/ctag-fh-kiel/ctag-face-2-4/files/980860/output.txt)

So if I install an official BeagleBoard.org image, I can use the functions of the AD1938 out of the box with the commands described on this repository?

Is that correct?

**Comment by henrix** [comment]
Have you generated a buildconfig (bb.org_defconfig) before compilation?
I just checked it and I don't get those errors.

Yes, after you flashed a recent image to a SD-card, you have to set the base device tree in /boot/uEnv.txt (dtb=am335x-bonegreen-ctag-face.dtb) and compile the appropriate device tree overlay (e.g. BB-CTAG-FACE-8CH-00A0.dts) from this repo using device tree compiler (e.g. dtc -O dtb -o BB-CTAG-FACE-8CH-00A0.dtbo -b 0 -@ BB-CTAG-FACE-8CH-00A0.dts) and move it to /lib/firmware as described in the [kernel compilation guide](https://github.com/ctag-fh-kiel/ctag-face-2-4/blob/master/docs/Kernel_Compilation_Guide.md). 

Then you can simply load the overlay and the soundcard is showed up with aplay -l.

**Comment by thmsmlt** [comment]
Yes I did, I just repeated all of the steps and at one point I got a prompt to install this thing: 'sudo apt-get install tor-arm', which I did. Then I ran the entire routine again and surely the error message was gone on the first compilation command, but afterwards, my terminal started throwing other errors. For future reference/improvement/fixing I'll attach the full terminal conversation to this post, but at this time, I'll be moving on using an official BeagleBoard.org image.

[Full output.txt](https://github.com/ctag-fh-kiel/ctag-face-2-4/files/983086/Full.output.txt)
